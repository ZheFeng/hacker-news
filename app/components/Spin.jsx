// @flow
import React, { Component } from 'react';

const inner = {
  borderRadius: '100%',
  transition: 'all 1000ms cubic-bezier(0.23, 1, 0.32, 1) 0ms',
  // position: 'absolute',
  width: '32px',
  height: '32px',
  border: '2px solid #0275d8',
  borderTopColor: 'transparent',
  transform: 'rotate(0deg)',
  margin: '0 auto',
};

class Spin extends Component {
  state = {
    rotate: 0,
  }
  componentDidMount() {
    this.timer = setTimeout(this.setPercentage.bind(this), 100);
  }
  componentWillUnmount() {
    clearTimeout(this.timer);
  }
  setPercentage() {
    const { rotate } = this.state;
    return this.setState(
      { rotate: rotate + 180 },
      () => this.timer = setTimeout(this.setPercentage.bind(this), 1000),
    );
  }
  render() {
    const { rotate } = this.state;
    const style = { ...inner, transform: `rotate(${rotate}deg)` };
    return (
      <div><div style={style} /></div>
    );
  }
  timer = null
}


// ContentItem.propTypes = {
//   title: PropTypes.string.isRequired,
//   time: PropTypes.number,
//   text: PropTypes.string,
//   url: PropTypes.string,
// };
// ContentItem.defaultProps = {
//   time: null,
//   text: null,
//   url: null,
// };

export default Spin;
