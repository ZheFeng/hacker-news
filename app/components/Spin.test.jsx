/* eslint-disable import/no-extraneous-dependencies, max-nested-callbacks */
/* global describe, it, expect */
import React from 'react';
import renderer from 'react-test-renderer';

import Spin from './Spin';

describe('<Spin />', () => {
  it('Snapshot Test', () => {
    const component = renderer.create(
      <Spin />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
