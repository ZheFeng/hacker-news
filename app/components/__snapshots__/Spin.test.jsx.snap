exports[`<Spin /> Snapshot Test 1`] = `
<div>
  <div
    style={
      Object {
        "border": "2px solid #0275d8",
        "borderLeftColor": "transparent",
        "borderRadius": "100%",
        "borderRightColor": "transparent",
        "height": "32px",
        "margin": "0 auto",
        "transform": "rotate(0deg)",
        "transition": "all 1000ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": "32px",
      }
    } />
</div>
`;
